src/analyze-dependencies.ts

```diff
@@ -0,0 +1,273 @@
+      throw error;
+    }
+  }
+
+  private async walkNodeModules(
+    dir: string,
+    callbacks: {
+      onPackage: (pkgJson: any) => void;
+      onFile: (filePath: string) => void;
+    },
+    seenPackages = new Set<string>()
+  ) {
+    try {
+      const entries = await fs.readdir(dir, {withFileTypes: true});
+
+      for (const entry of entries) {
+        const fullPath = path.join(dir, entry.name);
+
+        // Handle symlinks
+        if (entry.isSymbolicLink()) {
+          try {
+            const realPath = await fs.realpath(fullPath);
+            logger.debug('Found symlink:', fullPath, '->', realPath);
+            // If the real path is a package, process it
+            const pkgJsonPath = path.join(realPath, 'package.json');
+            try {
+              const pkgJson = JSON.parse(
+                await fs.readFile(pkgJsonPath, 'utf-8')
+              );
+              if (!seenPackages.has(pkgJson.name)) {
+                seenPackages.add(pkgJson.name);
+                logger.debug('Detected package (symlink):', pkgJson.name, 'at', realPath);
+                callbacks.onPackage(pkgJson);
+              } else {
+                logger.debug('Already seen package (symlink):', pkgJson.name, 'at', realPath);
+              }
+            } catch {
+              // Not a package or can't read package.json, continue
+            }
+            // Only follow symlinks that point to node_modules
+            if (realPath.includes('node_modules')) {
+              logger.debug('Following symlink to:', realPath);
+              await this.walkNodeModules(realPath, callbacks, seenPackages);
+            }
+          } catch {
+            logger.debug('Error resolving symlink:', fullPath);
+          }
+          continue;
+        }
+
+        if (entry.isDirectory()) {
+          // Check if this is a package directory
+          const pkgJsonPath = path.join(fullPath, 'package.json');
+          try {
+            const pkgJson = JSON.parse(await fs.readFile(pkgJsonPath, 'utf-8'));
+            // Only process each package once
+            if (!seenPackages.has(pkgJson.name)) {
+              seenPackages.add(pkgJson.name);
+              logger.debug('Detected package:', pkgJson.name, 'at', fullPath);
+              callbacks.onPackage(pkgJson);
+            } else {
+              logger.debug('Already seen package:', pkgJson.name, 'at', fullPath);
+            }
+          } catch {
+            // Not a package or can't read package.json, continue walking
+          }
+
+          // Continue walking if it's not node_modules
+          if (entry.name !== 'node_modules') {
+            await this.walkNodeModules(fullPath, callbacks, seenPackages);
+          }
+        } else {
+          callbacks.onFile(fullPath);
+        }
+      }
+    } catch {
+      logger.debug('Error walking directory:', dir);
+    }
+  }
+}
+
+export class RemoteDependencyAnalyzer implements DependencyAnalyzer {
```

Comment:
is this unused now? we should probably just remove it until we get chance to implement it properly